ckan/logic/action/create.py

```diff
@@ -286,6 +286,9 @@ def resource_create(context, data_dict):
     if not data_dict.get('url'):
         data_dict['url'] = ''
 
+    # lock the package record in the db when selected via package_show()
+    context['for_update'] = True
+
     pkg_dict = _get_action('package_show')(
         dict(context, return_type='dict'),
```

Comment:
Committed another version where I modified the context just for the specific package_show() action ( same for both cases: update and create )